fmc/src/flow/rt_alias.rs

```diff
@@ -161,12 +162,25 @@ impl RtAliasLayer {
     /// * `env` - FMC Environment
     /// * `hand_off` - HandOff
     pub fn extend_pcrs(env: &mut FmcEnv) -> CaliptraResult<()> {
-        match env.soc_ifc.reset_reason() {
-            ResetReason::ColdReset | ResetReason::UpdateReset => extend_pcr_common(env),
-            _ => {
+        let reset_reason = env.soc_ifc.reset_reason();
+        match reset_reason {
+            ResetReason::ColdReset => {
+                cfi_assert_eq(reset_reason, ResetReason::ColdReset);
+                extend_pcr_common(env)
+            }
+            ResetReason::UpdateReset => {
+                cfi_assert_eq(reset_reason, ResetReason::UpdateReset);
+                extend_pcr_common(env)
+            }
+            ResetReason::WarmReset => {
+                cfi_assert_eq(reset_reason, ResetReason::WarmReset);
                 cprintln!("[alias rt : skip pcr extension");
                 Ok(())
             }
+            ResetReason::Unknown => {
```

Comment:
This is a change in FMC behavior; previously it would have just continued booting. Maybe this is fine, just calling it out.

Reply:
Where is this requirement coming from?

Reply:
We error on UnknownReset in ROM and RT so I was thinking we should also do the same in FMC. I can change it back if it's undesired behavior though.

Reply:
That's fine. ROM will catch the unknown reset before handing it off. The main goal is to harden the FMC.

Reply:
So should we error on UnknownReset in FMC or go back to how we were handling it before?

Reply:
It would be nice to spell out the criteria for adding the extra CFI given the penalty in code size.

Reply:
It is ok to add the extra error handling given the overall goal is to harden the FMC.